src/common/JSHandle.ts

```diff
@@ -1069,6 +1073,9 @@ export class ElementHandle<
   ): Promise<ElementHandle<T> | null> {
     const { updatedSelector, queryHandler } =
       getQueryHandlerAndSelector(selector);
+    if (!queryHandler.queryOne) {
```

Comment:
why not assert like for queryAllArray?

Reply:
This is more robust. Do we want to error? What's the expectation?

Reply:
why is it different from the handling of queryAllArray below?

Reply:
For that specific instance, there is no canonical choice for a return value, so I assert there.